### PR TITLE
291 - Integrate Slack API to the experiment data form for posting the notification in slack

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number
Closing #291

## Purpose/Implementation Notes
The following changes were made:
 - integrated the Slack API to the experiment data form in order to post users' request data submission to slack
 - replace the experiment request button with the text provided in the issue after users submission

Please see the latest UI [here](https://refinebio-7afep26yh-ccdl.vercel.app/).

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/shared/SearchCard/SearchCardAction/RequestExperimentFormButton.js [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
 import { RequestExperimentForm } from './RequestExperimentForm'
 
 export const RequestExperimentFormButton = ({ accessionCode }) => {
+  const { requestedExperiments, setRequestedExperiments } = useRefinebio()
   const { closeModal, openModal } = useModal()
   const id = `request-experiment-form-${accessionCode}`
 
+  const addRequestedExperiment = () => {
+    setRequestedExperiments([...requestedExperiments, accessionCode])
+  }
```
Comment (davidsmejia) [inline src/components/shared/SearchCard/SearchCardAction/RequestExperimentFormButton.js:15]: Can't we do this inside of the component? I think this should be defined in the context and exported.

**Review by davidsmejia — COMMENTED** [review]
Just need to clear up where we define `addRequestedExperiment`.

File: src/components/shared/SearchCard/SearchCardAction/RequestExperimentFormButton.js (hunk 1 above)
Comment (nozomione) [reply src/components/shared/SearchCard/SearchCardAction/RequestExperimentFormButton.js]: `addRequestedExperiment` is now defined inside the `RequestExperimentForm` component at 9035075f956fd9fe92334cc17a1243b16badcb57 👍

File: src/components/shared/SearchCard/SearchCardAction/RequestExperimentForm.js [hunk 2]
```diff
@@ -21,17 +28,24 @@ export const RequestExperimentForm = ({ accessionCode, closeForm }) => {
...
           })
-          // eslint-disable-next-line no-console
-          console.log(values)
+
+          // adds the requested experiment's accession code if not 500
+          if (response.status !== 500) {
+            addRequestedExperiment()
+          }
+
+          closeForm()
```
Comment (davidsmejia) [inline src/components/shared/SearchCard/SearchCardAction/RequestExperimentForm.js:48]: If we pass in the `closeForm` as a property, we should have the prop default to noop. I think it makes more sense to call it `onSubmit`

Comment (nozomione) [reply src/components/shared/SearchCard/SearchCardAction/RequestExperimentForm.js]: See the commit at 9913608ddfd2aa973a3fb2439ef7c9281d3c2689.

**Review by davidsmejia — APPROVED** [review]
LGTM

**Pull request merged**